### DISCUSSION TITLE
Hello from Iran

### DISCUSSION THREAD

**Discussion by MortThePersian** [XTLS/Xray-core#1921] [Q&A]
Dear Sisters & Brothers;

Please accept Our warmest regards from Iran.

Your legacy is already fulfilled, While the regime implemented extreme restrictions over the Internet here & even well-known names in VPN industry have become nearly obsolete v2ray has become Our saviour !

We need your help more than ever. 
Restrictions are becoming even more tight & it will indeed need your attention !

Currently We use VMess, Trojan & Vless over WS & gRPC plus Cloudflare CDN & a well working Cloudflare ip(that is not blocked or limited).
We also use WARP tools to overcome some problems such as repeating Google reCAPTCHA issue.

But everything is collapsing & with Our various testing Vision + Reality shown some promises.

Some questions rised that nobody seems to have a clear answer for & researching also had lead to nowhere ...
Can you please kindly explain some of these questions & make things more clear ?!

- Does Xray-core removing Vless ?

- What protocol you recommend most & why ? for example VMess looks absolutely ok but it's time dependant & We faced problem with it due to Day light saving time 

- What transmission protocol you recommend the most ? since TCP & CDN can't be used together & Vision + Reality is based on TCP

- Is XTLS will be available & remain in Xray-core  or it will be removed ? & do you recommend to use it over TLS & why ?

Sincerely

**Comment by nursery01** [comment]
> Does Xray-core removing Vless ?

Why?

> What protocol you recommend most & why ? 

Different technologies perform differently in different places. Try more. You should use different technologies to decentralise features.

> since TCP & CDN can't be used together & Vision + Reality is based on TCP

??? Ws and GRPC is also based on tcp.

**Reply by MortThePersian** [reply]
Hi, 

> Why?

That's also My question & many others, Removing Vless rumours started from v2fly Vless page on their official website as long as I know !
https://www.v2fly.org/en_US/v5/config/proxy/vless.html

> Different technologies perform differently in different places. Try more. You should use different technologies to decentralise features.

Well, As I mentioned VMess for example that have time dependency & Vless inherently have no any built-in security option (from v2fly official website)
I was curious is there a comparison or benchmark or maybe developer's opinion what are pros & cons of each of these protocols ?!

> ??? Ws and GRPC is also based on tcp.

We are currently have no choice but to use Cloudflare, But We have hopes in regards to Vision + Reality, If this makes clear what I meant !

**Reply by nursery01** [reply]
> I was curious is there a comparison or benchmark or maybe developer's opinion what are pros & cons of each of these protocols ?!

If you use vless+cdn, then your data can be obtained by cdn.

> We are currently have no choice but to use Cloudflare, But We have hopes in regards to Vision + Reality, If this makes clear what I meant !

Vision + Reality, working principle is different. I think it can never use cdn. unless it discards its original function and redesigns it.
If its functions and disadvantages become the same as ws+cdn, then it makes no sense.

The difficult point is "TLS in TLS" and TLS fingerprints of camouflage websites cannot be copy.

**Reply by bart3nder** [reply]
“ If you use vless+cdn, then your data can be obtained by cdn.”
How? Correct me if i’m wrong but as far as i know, if you route traffic through a safe channel like tls, it’s encrypted and only the final destination has the keys to decrypt it. it’s true that cdn obtain data and traffic, but it’s useless since it’s encrypted, specially if you configure it to encrypt the traffic not only from client to cdn, but cdn itself to the final server.

**Reply by nursery01** [reply]
> How? Correct me if i’m wrong but as far as i know, if you route traffic through a safe channel like tls, it’s encrypted and only the final destination has the keys to decrypt it. it’s true that cdn obtain data and traffic, but it’s useless since it’s encrypted, specially if you configure it to encrypt the traffic not only from client to cdn, but cdn itself to the final server.

Yes,you are right. But the dns query may be leaked. If the target website uses 80 (http), then the CDN can completely obtain the data. If you are using cloudflare then this question is not important. If you are using a Chinese or Iranian cdn then this is a big problem

**Reply by bart3nder** [reply]
Sorry i don’t get it, How is dns leak related and dangerous? I only consider it as an issue when user is dealing with sensitive works which may cause problems if they detect you are using vpn, like transactions, trading or …

**Reply by nursery01** [reply]
> Sorry i don’t get it, How is dns leak related and dangerous? I only consider it as an issue when user is dealing with sensitive works which may cause problems if they detect you are using vpn, like transactions, trading or …

Some Chinese use Chinese companies' cdn (USA IP or others country IP).
If use vless+cdn, then this line may be blocked
Google is a sensitive website in China, and facebook and twitter and news site and more......

**Comment by wangdaye078** [comment]
cdn + caddy + trojan(vmess or vless, trojan maybe better) + grpc(ws, grpc is better)

**Reply by RPRX** [reply]
**VLESS is better than Trojan, especially after VLESS has the enhanced version of XUDP as the default UoT solution.**

An article introducing the enhanced version of XUDP, and more articles coming soon...

**Reply by FranzKafkaYu** [reply]
> **VLESS is better than Trojan, especially after VLESS has the enhanced version of XUDP as the default UoT solution.**
> 
> An article introducing the enhanced version of XUDP, and more articles coming soon...

Can't wait to see…

**Comment by RPRX** [comment]
> Does Xray-core removing Vless ?

**No. Xray-core will NEVER remove VLESS.**

**On the contrary, we're continuously developing most advanced new features for the better use of VLESS, such as XTLS, REALITY, XUDP...**

> That's also My question & many others, Removing Vless rumours started from v2fly Vless page on their official website as long as I know ! https://www.v2fly.org/en_US/v5/config/proxy/vless.html

In China, almost everyone who knows *ray-core knows that **I'm the author of VLESS and there was a bad history between v2fly and me.**
So we all know that "v2fly is removing vless" is for political reasons and **no one takes it seriously because we use Xray-core.**

> Is XTLS will be available & remain in Xray-core or it will be removed ? & do you recommend to use it over TLS & why ?

**XTLS is evolved, not removed. We just changed the way to configure it (VLESS flow only), to make it available for both TLS and REALITY.**
**As REALITY is better than TLS in Xray-core, it's recommended to use [VLESS-XTLS-uTLS-REALITY](https://github.com/XTLS/REALITY#readme) now if you want to use XTLS Vision.**

**Reply by nursery01** [reply]
> I'm the author of VLESS and there was a bad history between v2fly and me.

發生了什麼原因，吵架咩？

**Reply by wangdaye078** [reply]
应该是当初几个作者理念上有了分歧，然后R大就独立出来做了XRAY。

**Reply by RPRX** [reply]
> > I'm the author of VLESS and there was a bad history between v2fly and me.
> 
> 發生了什麼原因，吵架咩？

~~原因很多维且复杂，你问原因才会导致吵架。~~

**Reply by MortThePersian** [reply]
> > Does Xray-core removing Vless ?
> 
> **No. Xray-core will NEVER remove VLESS.**
> 
> **On the contrary, we're continuously developing most advanced new features for the better use of VLESS, such as XTLS, REALITY, XUDP...**
> 
> > That's also My question & many others, Removing Vless rumours started from v2fly Vless page on their official website as long as I know ! https://www.v2fly.org/en_US/v5/config/proxy/vless.html
> 
> In China, almost everyone who knows *ray-core knows that **I'm the author of VLESS and there was a bad history between v2fly and me.** So we all know that "v2fly is removing vless" is for political reasons and **no one takes it seriously because we use Xray-core.**
> 
> > Is XTLS will be available & remain in Xray-core or it will be removed ? & do you recommend to use it over TLS & why ?
> 
> **XTLS is evolved, not removed. We just changed the way to configure it (VLESS flow only), to make it available for both TLS and REALITY.** **As REALITY is better than TLS in Xray-core, it's recommended to use [VLESS-XTLS-uTLS-REALITY](https://github.com/XTLS/REALITY#readme) now if you want to use XTLS Vision.**

Dear RPRX;

Thank you so much for the effort you put on this project. This project worth much more than you probably think about !
I can finally share with confident about these questions that rised debates.

I'm just encountered another thing I'd really like to know your opinion on that, What min & max tls version do you recommend ? Arguing over v1.3 specially have become confusing !

Sincerely

**Reply by RPRX** [reply]
@chika0801 这个帖子有大量的基本事实错误，并且在往另一边带节奏，你都放出来？

**Reply by chika0801** [reply]
已删了

**Reply by RPRX** [reply]
估计该论坛的那位新晋魔怔人就是看了这篇小作文，~~只可惜他来晚了两年，两年前他可以找到很多魔怔的小伙伴~~
